src/clr-addons/numericfield/numeric-field.ts

```diff
@@ -222,9 +222,11 @@ export class ClrNumericField implements OnInit, OnDestroy, AfterViewChecked {
           }
           indexDecimalSep = j;
         }
-        if (char === '0' && removeLeadingZeros && ((result.length === 0 && (j + 1) !== value.length) || result === NEGATIVE)) {
-          ignoredChars++;
-          continue;
+        if (char === '0' && removeLeadingZeros) {
```

Comment:
Code is ok.
But please add some documentation as it needs some time to understand.